File: code/sort/selection_sort.py
""" SORTING"""

def less(x_value, y_value):
    """
    requires Transitive(x and y)
    """
    return x_value < y_value

def argmin(array, **kwargs):
    """ requires Transitive(x and y) """
    start = kwargs.get('start', 0)
    end = kwargs.get('end', len(array))
    cmp = kwargs.get('cmp', less)
    min_value = array[start]
    index = start
    while start < end:
        if cmp(array[start], min_value):
            min_value = array[start]
            index = start
        start += 1
    return index

def selection_sort(array, **kwargs):
    """ requres Transitive(x and y) """
    start = kwargs.get('start', 0)
    end = kwargs.get('end', len(array))
    cmp = kwargs.get('cmp', less)
    while start < end:
        i = argmin(array, start=start, end=end, cmp=cmp)
        array[i], array[start] = array[start], array[i]
        start += 1

def main():
    """ main """
    x_array = list(range(5000))
    selection_sort(x_array, start=0)

if __name__ == '__main__':
    main()

File: code/sort/selection_sort.rb
#this is a an example of a sorting algorithm
def selection_sort(num_array)
  n = num_array.length
  j = 0
  while j < n - 1
    iMin = num_array[j]
    i = j+1
    while i < n
      if num_array[i] < iMin
        iMin = num_array[i]
      end
      puts num_array[i]
     i = i + 1
   end
   iMin_index = num_array.index(iMin)
   num_array[j], num_array[iMin_index] = num_array[iMin_index], num_array[j]
   j = j + 1
  end
  return num_array
end
